### PR TITLE
refactor!: rework workflow types for clarity

### PR DESCRIPTION
**Pull request by woyuen**
Previously there was only three types of workflow log:
- notification
- action
- runError

Any information concerning skipping or stopping the workflow step
was embedded into workflow log step object.

New types 'skipped' and 'stopped' are introduced to avoid checking
into this nested object.

Moreover, multiple variables had to be checked to determine
the cause of a log error. Now we distinguish two types of errors:
- preRunError (e.g. incorrect expressions), renaming from old runError
- "new" runError, replacing action with step.error = true

If the property `notifyUrl` of a workflow object is specified,
the object sent to the URL will have some property changes:
- `status` removed
- `type` added

### CODE REVIEW

File: src/services/workflow.js [hunk 1]
```diff
@@ -94,11 +94,27 @@ function start ({ communication, serverPort }) {
...
+                // DEPRECATED:END
+            })
+            .orWhere(qb2 => {
+              return qb2
+                .where('type', 'runError')
+                .whereJsonSupersetOf('step', { handleErrors: true })
+            })
+        })
+      const startedACountQueryBuilder = queryBuilder.clone().whereIn('type',
+        ['action', 'skipped', 'stopped', 'runError']
```
Comment (abruere) [inline src/services/workflow.js]: Why not everything _but_ `notification` type to ensure we don’t forget anything?

File: src/services/workflow.js [hunk 2]
```diff
@@ -957,9 +974,10 @@ async function notifyAfterCompleted ({
 }) {
   const payload = {
     event: exposedEvent,
-    // if last step was successful, stopped/filtered or just skipped, this means the workflow was successful.
-    // An error interrupts the workflow.
-    status: lastLog.step.error ? 'error' : 'success',
```
Comment (abruere) [inline src/services/workflow.js:962]: Thanks for removing this redundant property of `notification` steps after discussion on slack.

Can you make sure it’s mentioned in commit message as well?

**Review by abruere — APPROVED** [review]
Looks good to me, thanks!

Please prepare [breaking change log (empty for now)](https://docs.api.stelace.com/?version=latest#7f785b07-7a40-4019-b571-473239ab7b8b) for API reference before merging, and include changes in future version draft.

File: src/services/workflow.js (hunk 1 above)
Comment (woyuen) [reply src/services/workflow.js]: Indeed there are many types. It is everything but `notification` and `preRunError`. I'll update with a `whereNotIn` with those two types, thanks.

File: src/services/workflow.js (hunk 2 above)
Comment (woyuen) [reply src/services/workflow.js:962]: Yes, sure !

**Pull request merged**